### ISSUE TITLE
Tokens not getting auto-refreshed

### ISSUE THREAD

**Issue by awalias** [supabase/gotrue-js#32]
reported by Omar:

"encountering forced logouts after ~30 min - 1 hour (not sure how long exactly)"

**Comment by thorwebdev** [comment]
@awalias that should have been fixed via #31 

Are they maybe using an outdated version?

**Comment by SweeToxin** [comment]
@awalias Thank you for the issue
@thorwebdev Ant just updated our "boxes" I'm waiting to see if it wasn't due to that before I can confirm
Thank you both for your time

**Comment by thorwebdev** [comment]
@SweeToxin the important thing is that you're using the latest version of `supabase-js` v.1.1.2

**Comment by awalias** [comment]
Confirmed supabase js is 1.1.2

**Comment by ykdojo** [comment]
@awalias @kiwicopple is this issue still active?

**Comment by kiwicopple** [comment]
Note from Omar

> we encounter this when running locally. i.e. on localhost when a token expires it just logs you off
> It seems from a cursory glance that token refreshing works fine on our deployed domains

I think in gotrue we need the ability to set multiple domain names where a token can be valid from (@awalias )

**Comment by kiwicopple** [comment]
Also to be investigated: https://github.com/supabase/gotrue-js/issues/54

**Comment by churichard** [comment]
I'm also running into this issue, or something similar. Specifically, my network requests return a `401` error, saying that the jwt has expired. I'm running a Next.js app on `localhost`, following this example for authentication: https://github.com/supabase/supabase/tree/master/examples/nextjs-with-supabase-auth

I don't actually get logged out, my network requests just start failing after a certain period of time. If I refresh the page, then I am somehow authenticated again and my network requests work fine. But I would expect the jwt to be auto-refreshed once it is expired, without me reloading the page.

**Comment by dommarr** [comment]
> I'm also running into this issue, or something similar. Specifically, my network requests return a `401` error, saying that the jwt has expired. I'm running a Next.js app on `localhost`, following this example for authentication: https://github.com/supabase/supabase/tree/master/examples/nextjs-with-supabase-auth
> 
> I don't actually get logged out, my network requests just start failing after a certain period of time. If I refresh the page, then I am somehow authenticated again and my network requests work fine. But I would expect the jwt to be auto-refreshed once it is expired, without me reloading the page.

+1 - same issue here

**Comment by awalias** [comment]
possibly related: https://github.com/supabase/gotrue-js/issues/61

**Comment by akiarostami** [comment]
@churichard did you fix this issue? I'm still seeing this problem, and it has made my app unusable. I would appreciate it if you could let us know the solution (if you fixed it).

**Comment by churichard** [comment]
@akiarostami I still run into [issues with auth](https://github.com/supabase/supabase-js/issues/254), albeit not the same one I ran into in the comment above. I think that one has been fixed.

If you're having issues, I'd recommend moving your authentication entirely client-side and **not** following the Next.js example (which uses SSR). The Next.js example didn't fully work last time I tried to do it; doing things client-side seems to work better.

**Comment by akiarostami** [comment]
Thanks @churichard for the quick reply. 

There are some issues with moving the authentication fully to the client side as well. In that case, we won't be able to use Next.js's server side features and would have to do all authentication checks, data loading, etc. on the client side as well. 

I'm currently doing some stuff client-side, but I'm still sporadically getting jwt expiration error when I call Supabase's methods. Don't you see any of that?

I hope the wonderful Supabase team pays some attention to this, as it makes the Next.js experience very unpleasant, if not unusable.

**Comment by churichard** [comment]
@akiarostami Sorry, I'm not running into any jwt expiration error issues. Maybe you could shoot them an email about it, or open a new issue.

**Comment by devtelec** [comment]
I am experiencing this as well.  Fully client side.  JWT token expires if app tab is left open for a while, and everything rendered conditionally based on existence of the session disappears in my UI (of course).  Upon refreshing the page, everything works again.

Wondering why this was closed if the issue persists?  Is there a fix I'm now aware of?

**Comment by silentworks** [comment]
@sjc5 this issue was fixed from what I can remember.

**Comment by rutikwankhade** [comment]
I am also facing the same issue. Has anyone found a fix around this?

**Comment by silentworks** [comment]
@rutikwankhade please open a new issue with the problem you are facing. Also outline how to replicate the problem in your new issue. This was fixed and this issue is closed.

**Comment by dthyresson** [comment]
Hi @rutikwankhade I just posted and Issue here https://github.com/supabase/gotrue-js/issues/201 that perhaps reproduces your case -- or adds some context.

TLDR; you can get into a state where a user has no more valid refresh tokens and then the client can no longer get a new access token -- even though it does try to refresh periodically.

If you see similar behavior, please add to the comments on that Issue to track. Thanks!
